### PR TITLE
Ignore RO file systems reboot script write and delete errors

### PR DESCRIPTION
**Pull request by KwisatzHaderach**
When running on RO file systems (as a target) tmt fails to write reboot
scripts to /usr/bin and fails, however the tests can easily run without them.
Another exception is added to rm command ran afterwards.

### CODE REVIEW

File: tmt/steps/execute/internal.py [hunk 1]
```diff
@@ -226,13 +226,21 @@ def _setup_reboot(self, guest):
...
+                        raise error
             yield
         finally:
             self.debug("Remove our reboot script implementations.", level=2)
             for reboot_file in REBOOT_SCRIPT_PATHS:
-                guest.execute(f'rm "{reboot_file}"')
+                try:
+                    guest.execute(f'rm "{reboot_file}"')
+                except tmt.utils.RunError as error:
+                    if error.returncode != 1:
```
Comment (psss) [inline tmt/steps/execute/internal.py:242]: Why are you checking specifically for exit code `1` here? In the docs I see

```
An exit status of zero indicates success, and a nonzero value indicates failure.
```

Is the status `1` defined for certain types of errors? Or should we just ignore unsuccessful removals?

**Review by psss — COMMENTED** [review]
Thanks for improving this. Looks good to me in general. Just not sure about the exit code check.

File: tmt/steps/execute/internal.py (hunk 1 above)
Comment (KwisatzHaderach) [reply tmt/steps/execute/internal.py:243]: RC 1 should be permission denied OS error but I would guess most other errors would also be indicated by 1.
However I wanted to keep the smallest impact I could, but of course could also be just `except: pass`.

Comment (psss) [reply tmt/steps/execute/internal.py:243]: Yeah, I've tried it. Seems the exit code `1` is used for non-existent files as well. So... probably does not make sense to check specifically for it. Let's go with `except: pass` unless anyone has objections. @FrNecas thoughts?

Comment (juanje) [reply tmt/steps/execute/internal.py:243]: @psss It makes sense to me what you're saying.

Could we add this change so we unblock the tests on OSTree based images? I guess we could try to find a better solution, but right now this is blocking all our tests :disappointed: 
Thanks!

**Review by psss — APPROVED** [review]
Thanks for the adjustments, looks good now.

**Pull request merged**